File: src/DataStructure/LinkedList/Medium/MyLinkedList/python/testCases.py
import json

try:
    data = open('testCases.json')
except FileNotFoundError:
    from pathlib import Path
    data = (Path(__file__).parent / '../testCases.json').open()

cases = json.load(data)


def runTestCase(MyLinkedList, testCase):
    list = MyLinkedList()
    commands = testCase['input'][1:]
    args = testCase['args'][1:]
    return [None] + [
        getattr(list, command)(*args[i])
        for i, command in enumerate(commands)
    ]

File: src/DataStructure/LinkedList/Medium/MyLinkedList/Singly/javascript/testCases.js
import json from '../../testCases.json';

export const cases = json;

export default (LinkedList, { input, args }) => {
  const list = new LinkedList();
  return input.map((command, i) => {
    const func = LinkedList.prototype[command];
    if (func) return func.apply(list, args[i]);
    return null;
  });
};
